packages/transaction-controller/src/helpers/PendingTransactionTracker.ts

```diff
@@ -295,7 +313,7 @@ export class PendingTransactionTracker {
   async #checkTransaction(txMeta: TransactionMeta) {
     const { hash, id } = txMeta;
 
-    if (!hash) {
+    if (!hash || !this.#beforeCheckPendingTransaction(txMeta)) {
```

Comment:
The extension logic appears to be skipping the no hash error if this hook returns `false`.

So should this condition instead be `!hash && this.#beforeCheckPendingTransaction(txMeta)` instead?

That way we only throw if there is no hash and we're not MMI mode?

Reply:
indeed the extension is the opposite, I applied your suggestion to fix it.